docs/KernelFeatures/UsingSecureVault.md

```diff
@@ -0,0 +1,108 @@
+
+WSO2 Carbon is shipped with a Secure Vault implementation. This allows you to store encrypted passwords that are mapped to aliases, i.e., you can use the aliases instead of the actual passwords in your configuration files for better security. For example, some configurations require the admin username and password. If the admin user password is 'admin', you could use the alias `UserManager.AdminUser.Password` in your configuration file. You would then map that alias to the actual password 'admin'. At runtime, the product will look up this alias in the secure vault and SecureVault will return the decrypted password.
+
+There are three files that are needed by the SecureVault:
+
+1. Secure-vault.yaml: 
+    Configurations that are required for configuring the SecureVault are given in this file.
+2. Master-keys.yaml: 
+    The default SecureVault implementation is based on the Java Key Store (JKS). Passwords that are needed to access the JKS and Keys are specified in this file. 
+3. Secrets.properties: 
+    This file contains the alias with plain text or encrypted password.
+    Example:
+    
+        UserManager.AdminUser.Password=plainText ABC@123
+        UserManager.AdminUser.Password=cipherText SnBSWKjtZZOo0UsmOpPRhP6ZMNYTb80+BZHRDC/kxNT9ExcTswAbFjb/aip2KgQNaVuIT27UtrBaIv77Mb5sNPGiwyPrfajLNhSOlke2p8YmMkegx/mG2ytJhJa5j9iMGtCsbMt+SAf85v6kGIiH0gZA20qDZ9jnveT7/Ifz7v0\=
+
+The SecureVault reads the aliases and passwords given in the secrets.properties file and return the resolved (decrypted) password.
+
+The SecureVault implementation has two major sub-components, namely the Master Key Reader and Secret Repository. The SecureVault implementation allows you to plugin custom implementations for both these sub-components:
+
+1. Secret Repository
+   The default implementation of SecretRepository is based on the passwords and aliases given in the Secrets.properties file and the default JKS that is packed with the Carbon product.
+2. Master Key Reader
+   The default implementation of MasterKeyReader gets a list of required passwords from the SecretRepository and provides the values for those passwords by reading system properties, environment variables and the master-keys.yaml file.
+
+## How To Use Secure Vault
+SecureVault is by default enabled. It reads the aliases and passwords given in the secrets.properties file. The secrets.properties file may contain both plain text and encrypted passwords. We have a separate tool called 'ciphertool' (shipped with Carbon /bin) to encrypt the secrets.properties file. Once the tool is run, it will encrypt all the plain text passwords in the secrets.properties file.
+
+CipherTool also depends on the configurations given in the secure-vault.yaml. Hence making necessary changes in the `secure-vault.yaml` file before running the CipherTool is mandatory. Once configured, running the 'ciphertool' is as simple as running the ciphertool script (ciphertool.sh on Linux/Mac and cihpertool.bat on Windows).
```

Comment:
secure-vault.yaml file. Therefore, it is mandatory to make changed in the `secure-vault.yaml` file before running the Cipher tool.